### PR TITLE
Aws email

### PR DESCRIPTION
**Pull request by RobStallion**
uses aws for sending forgotten password email #76
update tests
fix typo #270

### CODE REVIEW

**Comment by RobStallion** [comment]
currently having an issue getting travis to pass. Please see issue #272

File: test/controllers/user_controller_test.exs [hunk 1]
```diff
@@ -22,4 +22,30 @@ defmodule Bep.UserControllerTest do
...
+    conn = post conn, user_path(conn, :create, %{"user" => %{"email": "new@example.com", "password": "1"}})
+    assert html_response(conn, 200)
+  end
+
+  test "POST /users/create", %{conn: conn} do
+    insert_types()
+    conn = post conn, user_path(conn, :create, %{"user" => %{
+      "email": "new-email@example.com",
+      "password": "supersecret",
+      "1": "true"}})
```
Comment (shouston3) [inline test/controllers/user_controller_test.exs]: What is `"1": "true"`?

File: web/templates/user/new.html.eex [hunk 2]
```diff
@@ -8,7 +8,7 @@
     <div class="alert alert-danger">
       <p>Oops, something whent wrong!</p>
     </div>
-  <%= end %>
+  <% end %>
```
Comment (shouston3) [inline web/templates/user/new.html.eex]: 👍

File: lib/bep/mailer.ex [hunk 3]
```diff
@@ -0,0 +1,6 @@
+defmodule Bep.Mailer do
+  @moduledoc """
+  """
```
Comment (shouston3) [inline lib/bep/mailer.ex]: Potentially `@moduledoc false` could be here (not a big issue).

File: .travis.yml [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
   directories:
-    - _build
-    - deps
+  - _build
+  - deps
 addons:
   postgresql: '9.5'
 env:
-  - MIX_ENV=test
+- MIX_ENV=test
```
Comment (shouston3) [inline .travis.yml]: I think the correct way to do this is to define [global `env` variables](https://docs.travis-ci.com/user/environment-variables/#Global-Variables) like this:

```
env:
  global:
    - MIX_ENV=test
    - OTHER_VAR=other_var
```

What the current `.travis.yml` does is say I would like to run one build using this `env` (and potentially another build using another `env`). What the above code snippet says is: I would always like to run travis builds with these `env`s

File: .travis.yml [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
 before_script:
-  - mix do ecto.create, ecto.migrate
+- mix do ecto.create, ecto.migrate
 script:
-  - mix do deps.get, compile --warnings-as-errors, coveralls.json
+- MIX_ENV=test mix do deps.get, compile --warnings-as-errors, coveralls.json
```
Comment (shouston3) [inline .travis.yml]: you shouldn't need `MIX_ENV=test`, it's been defined in the vars above

File: test/controllers/user_controller_test.exs [hunk 6]
```diff
@@ -22,4 +22,30 @@ defmodule Bep.UserControllerTest do
...
+  end
+
+  test "POST /users/create redirect to /login when email already linked to an account", %{conn: conn} do
+    insert_user()
+    conn = post conn, user_path(conn, :create, %{"user" => %{"email": "email@example.com", "password": "supersecret"}})
+    assert html_response(conn, 302)
+  end
+
+  test "POST /users/create show /user/new when password is too short", %{conn: conn} do
+    insert_user()
```
Comment (shouston3) [inline test/controllers/user_controller_test.exs]: do we need this `inesrt_user()` for this test to pass?

Comment (RobStallion) [reply test/controllers/user_controller_test.exs]: Good spot 👀. I'll update that test

**Comment by RobStallion** [comment]
Those comments were all helpful thanks. Should all be resolved now @shouston3

File: lib/email.ex [hunk 7]
```diff
@@ -0,0 +1,15 @@
+defmodule Bep.Email do
+  @moduledoc """
+  """
```
Comment (shouston3) [inline lib/email.ex]: 🚩

File: test/support/mock_httpoison.ex [hunk 8]
```diff
@@ -1,4 +1,6 @@
 defmodule Bep.Mock.HTTPoison do
+  @moduledoc """
+  """
```
Comment (shouston3) [inline test/support/mock_httpoison.ex]: 🚩

**Comment by RobStallion** [comment]
@shouston3 Thanks for pointing that out. Updated again 👍

**Pull request merged**